src/manifolds/Stiefel.jl

```diff
@@ -162,6 +162,56 @@ function exp!(::Stiefel{n,k}, q, p, X) where {n,k}
     )
 end
 
+@doc raw"""
+    get_basis(M::Stiefel{n,k,ℝ}, p, B::DefaultBasis) where {n,k}
+
+Create the default basis using the parametrization for any $X ∈ T_p\mathcal M$.
+Set $p_\bot \in ℝ^{n\times(n-k)}$ the matrix such that the $n\times n$ matrix of the common
+columns $[p\ p_\bot]$ is an ONB.
+For any skew symmetric matrix $a ∈ ℝ^{k\times k}$ and any $b ∈ ℝ^{(n-k)\times k}$ the matrix
+
+````math
+X = pa + p_\bot b ∈ T_p\mathcal M
+````
+
+and we can use the $\frac{1}{2}k(k-1) + (n-k)k = nk-\frac{1}{2}k(k+1)$ entries
+of $a$ and $b$ to specify a basis for the tangent space.
+using unit vectors for constructing both
+the upper matrix of $a$ to build a skew symmetric matrix and the matrix b, the default
+basis is constructed.
+"""
+function get_basis(::Stiefel{n,k,ℝ}, p, B::DefaultBasis{ℝ}) where {n,k}
+    p⊥ = nullspace([p zeros(n, n - k)])
+    an = div(k * (k - 1), 2)
+    bn = (n - k) * k
+    V = [
+        [p * _vec2skew(_euclidean_unit_vector(an, i), k) for i in 1:an]...,
+        [p⊥ * reshape(_euclidean_unit_vector(bn, j), (n - k, k)) for j in 1:bn]...,
+    ]
+    return CachedBasis(B, V)
+end
+
+_euclidean_unit_vector(n, i) = [k == i ? 1.0 : 0.0 for k in 1:n]
+"""
+    vec2skew(v,k)
+
+create a skew symmetric matrix from a vector, for example for `v=[1,2,3]` and `k=3` this
+yields
+````julia
+[  0  1  2;
+  -1  0  3;
+  -2 -3  0
+]
+````
+"""
+function _vec2skew(v, k)
```

Comment:
Can this be unified with the machinery used by `Rotations` to make a skew-symmetric matrix?

Reply:
In `Rotations.jl` I only saw a 4d angles to skew matrix conversion, this takes a vector puts it in the upper triangular (without signal) part and then generates the corresponding skew symmetric matrix. If you can point out what exactly can be unified, I'd happily do that.
When used with more than one manifold, we surely also can put this function into the `utils.jl`.

Reply:
Ah, it's the implementation of `get_vector!`:
https://github.com/JuliaManifolds/Manifolds.jl/blob/5ad6f46523b3516978bc67723fcd3d64829ea464/src/manifolds/Rotations.jl#L318-L342
The one difference is this generates the upper 3x3 matrix as customary for `so(3)` and then fills the upper triangle for the rest, negating for the lower triangle.

An argument for keeping the same function is that `Rotations` is a submanifold of `Stiefel`. An argument for keeping them separate is that perhaps for `Stiefel` one would not want to use a convention like this that privileges a certain submanifold.

Reply:
Ah, sure I can use the new function for those. I will provide also a mutating variant then.

Reply:
Hm, looking closely, I am not sure why you exchange the third and second entry  (lines 323&324) compared to linearly writing them in. This currently makes the tests fail. Can you elaborate on this specific order?

Reply:
To be precise for the case of dimension 3 and a `v=[a,b,c]` I do
````julia
[ 0 a b;
  -a 0 c;
 -b -c 0 ]
````
and you currently (also in this sub part for larger ones)
````julia
[ 0 -c b;
  c 0 -a;
 -b a 0 ]
````
so if this is a convention usually used for SO(n) and its first 3 entries, we can not use my method here.

Reply:
Yes, this is the `so(3)` convention, chosen because the resulting 3-vector is the axis of rotation multiplied by the angle of rotation for the rotation one would get following the geodesic from the identity element in the direction of the corresponding tangent vector for unit time. Likewise, if only the upper 2x2 matrix is non-zero, then the result is a rotation in the xy-plane. There's perhaps a way to generalize the choice of sign convention for SO(n), but I haven't seen it.

Reply:
Thanks for the explanation, but then we can not easily unify both; however, we now have `vec2skew` and ´vec2skew!` available.

Reply:
Isn't SO(n) only diffeomorphic (not isomorphic) to a case of Stiefel with the Euclidean metric? If so I don't see much point in making vectorized representations of skew-symmetric matrices identical for both manifolds.

Reply:
That's also my conclusion, we should stick to the default SO(n) uses and the one Stiefel uses; sure they are diffeomorphic and there is no point enforcing them to be identical.